pubsub/google/cloud/pubsub_v1/subscriber/policy/base.py

```diff
@@ -361,22 +358,36 @@ def maintain_leases(self):
             p99 = self.histogram.percentile(99)
             _LOGGER.debug('The current p99 value is %d seconds.', p99)
 
+            # Drop any leases that are well beyond max lease time. This
+            # ensures that in the event of a badly behaving actor, we can
+            # drop messages and allow Pub/Sub to resend them.
+            cutoff = time.time() - self.flow_control.max_lease_duration
+            to_drop = [
+                DropRequest(ack_id, item.size)
+                for ack_id, item
+                in six.iteritems(self.leased_messages)
+                if item.added_time < cutoff]
+
+            if to_drop:
+                _LOGGER.warning(
+                    'Dropping %s items because they were leased too long.',
+                    len(to_drop))
+                self.drop(to_drop)
+
             # Create a streaming pull request.
             # We do not actually call `modify_ack_deadline` over and over
             # because it is more efficient to make a single request.
-            ack_ids = list(self.managed_ack_ids)
-            _LOGGER.debug('Renewing lease for %d ack IDs.', len(ack_ids))
+            ack_ids = list(self.leased_messages.keys())
             if ack_ids:
-                request = types.StreamingPullRequest(
-                    modify_deadline_ack_ids=ack_ids,
-                    modify_deadline_seconds=[p99] * len(ack_ids),
-                )
+                _LOGGER.debug('Renewing lease for %d ack IDs.', len(ack_ids))
+
                 # NOTE: This may not work as expected if ``consumer.active``
```

Comment:
Is this comment still true? It references `send_request`, which is no longer directly called.

Reply:
Yep, it is, even though `send_request` isn't called directly.